### PR TITLE
No `Map`s for you!

### PR DESCRIPTION
**Pull request by Centril**
# Description of Changes

Remove `AlgebraicType::Map` and `AlgebraicValue::Map` and all the consequence changes necessary.

# API and ABI breaking changes

Yes.

### CODE REVIEW

**Review by RReverser — APPROVED** [review]
I'm a simple man, I see someone removing tons of code, I click Approve.

File: crates/bindings-csharp/BSATN.Runtime/BSATN/Runtime.cs [hunk 1]
```diff
@@ -438,38 +438,3 @@ public AlgebraicType GetAlgebraicType(ITypeRegistrar registrar) =>
     public AlgebraicType GetAlgebraicType(ITypeRegistrar registrar) =>
         enumerable.GetAlgebraicType(registrar);
 }
-
-public readonly struct Dictionary<Key, Value, KeyRW, ValueRW> : IReadWrite<Dictionary<Key, Value>>
```
Comment (RReverser) [inline crates/bindings-csharp/BSATN.Runtime/BSATN/Runtime.cs:442]: @Centril Actually, can you also remove the relevant branch in BSATN.Codegen as well?

Comment (RReverser) [reply crates/bindings-csharp/BSATN.Runtime/BSATN/Runtime.cs:442]: I suppose C# tests will yell at you about it too.

**Review by bfops — APPROVED** [review]
CLI changes LGTM

**Pull request merged**